Automattic/woocommerce-payments
# Integrate Stripe payment method messaging into product details page

Fixes #6503 

#### Changes proposed in this Pull Request
This PR integrates the Stripe's [Payment Method Messaging Element](https://stripe.com/docs/payments/payment-method-messaging). This will allow displaying information related to the available BNPL providers in the product details page.

<img width="764" alt="Screenshot 2023-06-15 at 16 08 05" src="https://github.com/Automattic/woocommerce-payments/assets/1553182/83a57cce-6240-46e7-9ad0-d93aca7e8440">










#### Testing instructions





_Feature flag enabled and BNPL enabled_
1. Make sure that `is_bnpl_affirm_afterpay_enabled` is enabled. You can set the `bnpl_affirm_afterpay_enabled` meta to your account in the server (don't forget to clear the cache in the client)
2. Enable Affirm and Afterpay as available payment methods in the admin ( please check instructions in this [PR](https://github.com/Automattic/woocommerce-payments-server/pull/3405) to add fees
3. Navigate to the details page of a product. Check that there's a new block below the product that displays the BNPL information (see screenshot above)
4. Increment the quantity selector. Check that the information is updated in the BNPL block

_Feature flag enabled and BNPL disabled_
1. Make sure that `is_bnpl_affirm_afterpay_enabled` is enabled. 
2. Disable Affirm and Afterpay as available payment methods in the admin
3. Navigate to the details page of a product. Check that the BNPL block is not present

_Feature flag disabled_
1. Disable feature flag for the merchant.
5. Navigate to the details page of a product. Check that the BNPL block is not present
-------------------


**Post merge**
